# Comparison of functional normalization implementations

## Download example data set 

> Prickett AR, Ishida M, BÃ¶hm S, Frost JM et al. Genome-wide methylation
> analysis in Silver-Russell syndrome patients. Hum Genet 2015
> Mar;134(3):317-32. PMID: 25563730

Retrieve the data from the Gene Expression Omnibus (GEO) website
(http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE55491).

```{r}
dir.create(path <- "data")

if (length(list.files(path, "*.idat$")) == 0) {
  filename <-  file.path(path, "gse55491.tar")
  download.file("http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE55491&format=file", filename)
  cat(date(), "Extracting files from GEO archive.\n")
  system(paste("cd", path, ";", "tar xvf", basename(filename)))
  unlink(filename)
  cat(date(), "Unzipping IDAT files.\n")
  system(paste("cd", path, ";", "gunzip *.idat.gz"))
}
```

## Load and normalize using `minfi`

The current version `minfi::preprocessFunnorm()` contains a bug (Feb 25, 2015).
To make a proper comparison with `meffil`, this bug should be fixed
and `minfi` reinstalled.
This can be done with the included patch file: `fix-minfi-control-matrix.patch`.

Load the data.

```{r, message=FALSE}
library(minfi, quietly=TRUE)
```

Normalize using the minfi package.
```{r functional-normalization-minfi, cache=T}
raw.minfi <- read.450k.exp(base = path)
system.time(norm.minfi <- preprocessFunnorm(raw.minfi, nPCs=2,
                                            sex=NULL, bgCorr=TRUE, dyeCorr=TRUE, verbose=TRUE))
```

## Load and normalize using `meffil`

```{r}
options(mc.cores=10)
```

Load the code, probe annotation and sample filename information.
```{r functional-normalization-meffil-long, cache=T}
library(meffil)
samplesheet <- meffil.create.samplesheet(path)
meffil.get.cell.type.references()

qc.file <- "gse55491/qc-report.html"
author <- "Prickett, et al."
study <- "Silver-Russell syndrome patients (GEO:GSE55491)"
number.pcs <- 2
temp.dir <- "."
norm.file <- "gse55491/normalization-report.html"

qc.objects <- meffil.qc(samplesheet, cell.type.reference="blood gse35069", verbose=T)
qc.summary <- meffil.qc.summary(qc.objects, verbose=T)
meffil.qc.report(qc.summary,
                 output.file=qc.file,
                 author=author,
                 study=study)
if (nrow(qc.summary$bad.samples) > 0)
    qc.objects <- meffil.remove.samples(qc.objects, qc.summary$bad.samples$sample.name)
norm.objects <- meffil.normalize.quantiles(qc.objects, number.pcs=number.pcs, verbose=T)
norm.meffil <- meffil.normalize.samples(norm.objects,
                                        beta=F, 
                                        cpglist.remove=qc.summary$bad.cpgs$name,
                                        temp.dir=temp.dir,
                                        verbose=T)
beta.meffil <- meffil.get.beta(norm.meffil$M, norm.meffil$U)
norm.summary <- meffil.normalization.summary(beta.meffil, norm.objects=norm.objects, verbose=T)
meffil.normalization.report(norm.summary,
                            output.file=norm.file,
                            author=author,
                            study=study)
```

```{r functional-normalization-meffil-short, cache=T}
norm.meffil2 <- meffil.normalize.dataset(samplesheet,
                                        qc.file=qc.file,
                                        author=author,
                                        study=study,
                                        number.pcs=number.pcs,
                                        beta=T,
                                        temp.dir=temp.dir,
                                        norm.file=norm.file,
                                        verbose=T)
beta.meffil.2 <- norm.meffil2$data
```

```{r}
quantile(beta.meffil - beta.meffil.2[rownames(beta.meffil),])
```

## Compare normalizations

First make sure that the raw data is the same.
```{r}
raw.meffil <- meffil:::read.rg(samplesheet$Basename[1])
all(raw.meffil$R == getRed(raw.minfi)[names(raw.meffil$R),1])
```

### Background correction

Apply `meffil` background correction.
```{r, results='hide'}
bc.meffil <- meffil:::background.correct(raw.meffil)
M.meffil <- meffil:::rg.to.mu(bc.meffil)$M
```

Apply `minfi` background correction.
```{r, results='hide'}
bc.minfi <- preprocessNoob(raw.minfi, dyeCorr = FALSE)
M.minfi <- getMeth(bc.minfi)[names(M.meffil),1]
```

Compare results, should be identical.
```{r}
quantile(M.meffil - M.minfi)
```

### Dye bias correction
Apply `meffil` correction.
```{r, results='hide'}
dye.meffil <- meffil:::dye.bias.correct(bc.meffil,
                                        intensity=norm.objects[[1]]$reference.intensity)
M.meffil <- meffil:::rg.to.mu(dye.meffil)$M
U.meffil <- meffil:::rg.to.mu(dye.meffil)$U
```

Apply `minfi` background and dye bias correction.
```{r, results='hide'}
dye.minfi <- preprocessNoob(raw.minfi, dyeCorr = TRUE)
M.minfi <- getMeth(dye.minfi)[names(M.meffil),1]
U.minfi <- getUnmeth(dye.minfi)[names(U.meffil),1]
```

Compare results, should be identical.
```{r}
quantile(M.meffil - M.minfi)
quantile(U.meffil - U.minfi)
quantile(M.meffil/(M.meffil+U.meffil+100) - M.minfi/(M.minfi+U.minfi+100))
```

### Control matrix extracted

Extract `minfi` control matrix.
```{r, results='hide'}
library(matrixStats, quietly=TRUE)
control.matrix.minfi <- minfi:::.buildControlMatrix450k(minfi:::.extractFromRGSet450k(raw.minfi))
```

Control matrix for `meffil` was extracted earlier.
Preprocess the matrix using the same procedure used by `minfi`.
```{r, results='hide'}
control.matrix.meffil <- meffil.control.matrix(norm.objects)
control.matrix.meffil <- scale(t(control.matrix.meffil))
control.matrix.meffil[control.matrix.meffil > 3] <- 3
control.matrix.meffil[control.matrix.meffil < -3] <- -3
control.matrix.meffil <- scale(control.matrix.meffil)
```

The control variable order (columns) may be different between `minfi` and `meffil`
so they are reordered.
```{r}
i <- apply(control.matrix.minfi,2,function(v) {
    which.min(apply(control.matrix.meffil,2,function(w) {
        sum(abs(v-w))
    }))
})
control.matrix.meffil <- control.matrix.meffil[,i]
```

Compare resulting matrices, should be identical.
```{r}
quantile(control.matrix.meffil - control.matrix.minfi)
```

### Final beta values

```{r}
beta.minfi <- getBeta(norm.minfi)
quantile(beta.meffil - beta.minfi[rownames(beta.meffil),])
```

On what chromosomes are the biggest differences appearing?
```{r}
probes <- meffil.probe.info()
probe.chromosome <- probes$chr[match(rownames(beta.meffil), probes$name)]
sex <- sapply(norm.objects, function(object) object$predicted.sex)
is.diff <- abs(beta.meffil - beta.minfi[rownames(beta.meffil),]) > 1e-4
table(probe.chromosome[which(is.diff, arr.ind=T)[,"row"]])
table(probe.chromosome[which(is.diff[,which(sex=="M")], arr.ind=T)[,"row"]])
table(probe.chromosome[which(is.diff[,which(sex=="F")], arr.ind=T)[,"row"]])
```
These results not surprising
because chromosome Y not handled like `minfi` in either males or females, 
and chromosome X is handled just like `minfi` in females but not in males.

```{r}
autosomal.cgs <- unique(probes$name[which(probes$chr %in% paste("chr", 1:22, sep=""))])
autosomal.cgs <- intersect(autosomal.cgs, rownames(beta.meffil))
quantile(beta.meffil[autosomal.cgs,] - beta.minfi[autosomal.cgs,])
```

In spite of the differences, CG correlations between methods are pretty close to 1.
```{r}
male.idx <- which(rowSums(is.diff[,sex=="M"]) >= 5)
male.cg.r <- unlist(mclapply(male.idx, function(idx) {
    cor(beta.meffil[idx, sex=="M"], beta.minfi[rownames(beta.meffil)[idx], sex=="M"])
}))
quantile(male.cg.r, probs=c(0.05,0.1,0.25,0.5))

female.idx <- which(rowSums(is.diff[,sex=="F"]) > 0)
female.cg.r <- sapply(female.idx[1:200], function(idx) {
    cor(beta.meffil[idx, sex=="F"], beta.minfi[rownames(beta.meffil)[idx], sex=="F"])
})
quantile(female.cg.r, probs=c(0.05,0.1,0.25, 0.5))
```

## Sex

```{r}
sex.minfi <- as.data.frame(getSex(norm.minfi))
names(norm.objects) <- sapply(norm.objects, function(obj) basename(obj$basename))
sex.meffil <- data.frame(predicted=sapply(norm.objects, function(obj) obj$predicted.sex),
                         x.signal=sapply(norm.objects, function(obj) obj$x.signal),
                         y.signal=sapply(norm.objects, function(obj) obj$y.signal))
all(sex.minfi[rownames(sex.meffil),"predictedSex"] == sex.meffil$predicted)
quantile(sex.minfi[rownames(sex.meffil),"xMed"] - sex.meffil$x.signal)
quantile(sex.minfi[rownames(sex.meffil),"yMed"] - sex.meffil$y.signal)
```

## Cell count estimates

```{r, results="hide"}
counts.meffil <- t(meffil.cell.count.estimates(norm.objects))
```

```{r, results="hide"}
counts.minfi <- estimateCellCounts(raw.minfi)
counts.minfi <- counts.minfi[rownames(counts.meffil), colnames(counts.meffil)]
```

```{r}
sapply(rownames(counts.meffil), function(i) cor(counts.meffil[i,], counts.minfi[i,]))
sapply(colnames(counts.meffil), function(i) cor(counts.meffil[,i], counts.minfi[,i]))
```

```{r}
library(Cairo, quietly=TRUE)
```

```{r, dev="CairoPNG"}
for (cell.type in colnames(counts.meffil)) {
    lim <- range(c(counts.meffil[,cell.type], counts.minfi[,cell.type]))
    plot(counts.meffil[,cell.type], counts.minfi[,cell.type], pch=19,
         main=cell.type,
         sub=paste("R =", format(cor(counts.meffil[,cell.type], counts.minfi[,cell.type]), digits=3)),
         xlim=lim,ylim=lim)
    abline(a=0,b=1,col="red")
}
```

This file was generated from R markdown:

```{r, eval=F}
library(knitr)
knit("functional-normalization.rmd")
library(markdown)
markdownToHTML("functional-normalization.md","functional-normalization.html")
```
